libdevice/fallback-complex.cpp

```diff
@@ -150,8 +150,14 @@ float __complex__ __devicelib_cexpf(float __complex__ z) {
         z_imag = NAN;
       return CMPLXF(z_real, z_imag);
     }
-  } else if (__spirv_IsNan(z_real) && (z_imag == 0.0f)) {
-    return z;
+  } else if (__spirv_IsNan(z_real)) {
+    if (z_imag == 0.0f)
+      return z;
+    else /* z_imag != 0.0f */
+      return CMPLX(NAN, NAN);
```

Comment:
```suggestion
    /* z_imag != 0.0f */
    return CMPLX(NAN, NAN);
```
https://llvm.org/docs/CodingStandards.html#don-t-use-else-after-a-return

Reply:
Updated in [f898b54](https://github.com/intel/llvm/pull/15980/commits/f898b547b55d099ecedf1a47588a2efa139274df)